org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -186,11 +187,16 @@ public void registerCapabilities(InitializeResult initializeResult) {
 
 	@Override
 	public void triggerInitialization(Collection<IPath> roots) {
-		try {
-			// a workaround for https://github.com/redhat-developer/vscode-java/issues/2020
-			JobHelpers.waitForBuildJobs(2 * 60 * 1000); // 2 minutes
-		} catch (OperationCanceledException e) {
-			logException(e.getMessage(), e);
+		if (ProjectUtils.getAllProjects().length == 0) {
+			try {
+				// a workaround for https://github.com/redhat-developer/vscode-java/issues/2020
+				JavaLanguageServerPlugin.logInfo("Wait for AutoBuildOffJob start");
+				long start = System.currentTimeMillis();
+				JobHelpers.waitForBuildOffJobs(2 * 60 * 1000); // 2 minutes
```

Comment:
Just wanted to confirm : JobHelpers.waitForBuildJobs(..) had a pattern that was too general and would sometimes wait for build jobs as opposed to just the "build off" job that we wanted?

Reply:
Right.